orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -239,3 +239,37 @@ def test_model_to_device_and_back_to_original(original_device, to_device):
     assert model._device == torch.device(original_device+':0')
     for _, parameter_value in model.named_parameters():
         assert parameter_value.device.type == original_device
+
+def test_input_requires_grad_saved():
```

Comment:
This test is quiet simple and could be merged with the one below

Reply:
I actually require this test. Here I want to test that a non leaf tensor can be correctly deepcopied (which is why i use ```x = tensor(...)+1```).
And in the below test, I cannot use ```x = tensor(...)+1``` because I need to ```assert x.grad is not None```.